src/main/java/com/pinterest/secor/parser/Iso8601MessageParser.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.secor.parser;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import net.minidev.json.JSONObject;
+import net.minidev.json.JSONValue;
+import javax.xml.bind.DatatypeConverter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.pinterest.secor.common.SecorConfig;
+import com.pinterest.secor.message.Message;
+
+/**
+ * Iso8601MessageParser extracts timestamp field (specified by 'message.timestamp.name')
+ *
+ * @author Jurriaan Pruis (email@jurriaanpruis.nl)
+ *
+ */
+public class Iso8601MessageParser extends MessageParser {
+    private static final Logger LOG = LoggerFactory.getLogger(DateMessageParser.class);
+    protected static final String defaultDate = "dt=1970-01-01";
+    protected static final String defaultFormatter = "yyyy-MM-dd";
+
+    public Iso8601MessageParser(SecorConfig config) {
+        super(config);
+    }
+
+    @Override
+    public String[] extractPartitions(Message message) {
+        JSONObject jsonObject = (JSONObject) JSONValue.parse(message.getPayload());
+        String result[] = { defaultDate };
+
+        if (jsonObject != null) {
+            Object fieldValue = getJsonFieldValue(jsonObject);
+            if (fieldValue != null) {
```

Comment:
When fieldValue is not present, it will use the default, this might be error prone.  At least we need a warning logging here.

It might be better to introduce a config parameter to decide whether fallback to default or error out.

This also applies on when the field is available, but cannot be parsed.